Angular 7 example

ganny26 opened the pull request:
Hi @timothyis ,

Created a simple angular 7 app example with routing implemented

https://zeit-angular-app-gksnyrvbi.now.sh

#152

timothyis commented:
Hi @ganny26. Thank you so much for the submission and I apologise for the late response here. I will be testing this in a few hours and working to merge this in 🎉 

One topic before hand though: What do you think about extending this example to be a realistic project. For example; following the static example (https://github.com/zeit/now-examples/tree/master/static), where we pull in 5 Hacker News posts from their API and displaying them.

The example could even go further by having a post page showing comments, using the power of Angular.

Is this something you'd be interested in working on?

ganny26 commented:
Yeah sure @timothyis, Sounds interesting. I will update my PR with realistic project example.

ganny26 commented:
Hi @timothyis created a sample angular 7 Hacker News app, review and let me know your feedback 🙏

Demo Url : https://zeit-angular-app-exr7ho5vt.now.sh/

timothyis commented:
Amazing work @ganny26! I'll test and review this tomorrow!

timothyis commented:
@ganny26 Is there much difference between versions of Angular? For example, can we rename the directory to `angular` with the version and still be okay? Wondering if we really need an example for each version rather than maintaining one.

ganny26 commented:
@timothyis Each angular version release contains only core functional improvements and typescript version support changes not much breaking changes. I think it's better to keep folder name without version number will be okay ✌️

timothyis commented:
If you wouldn't mind remove the version from the directory and name, that would be great @ganny26! Then we can look at merging this in 🎉

ganny26 commented:
> If you wouldn't mind remove the version from the directory and name, that would be great @ganny26! Then we can look at merging this in 🎉

@timothyis Updated my PR with changes 👍

timothyis commented on angular/README.md:
````diff
@@ -0,0 +1,117 @@
+```
+
+cd ./src/app
+mkdir components
````
Seems like there's a missing step here. The user should cd into the directory, correct?

timothyis commented on angular/README.md:
````diff
@@ -0,0 +1,117 @@
+
+```
+
+Then add `<router-outlet></router-outlet>` to out root of AppComponent:
````
Did you mean to add an example here?

timothyis commented on angular/README.md:
```diff
@@ -0,0 +1,117 @@
+
+Create `.nowignore` and add `node_modules`
+
+We are now ready to deploy the app.
```
This readme omits the fact that the user needs a `now-build` script in the package.json

timothyis commented on angular/README.md:
````diff
@@ -0,0 +1,117 @@
+
+```
+
+It will generate three files `Markup,Component,StyleSheet`
````
What does this mean? I wasn't able to figure it out from trying this.

timothyis commented on angular/README.md:
````diff
@@ -0,0 +1,117 @@
+### Run app in development mode
+
+```
+yarn start
````
Perhaps we should note that Angular initiates the app with npm, e.g. package-lock.json, by default. And that a user will have to run `yarn` to get a yarn.lock file

timothyis commented on angular/README.md:
```diff
@@ -0,0 +1,117 @@
+
+cd ./src/app
+mkdir components
+ng generate component  --spec=false about
```
Now that I've done this, I get the following error:
```
ERROR in : Cannot determine the module for class AboutComponent in /tmp/2025acf/src/app/components/about/about.component.ts! Add AboutComponent to the NgModule to fix it.
Cannot determine the module for class HomeComponent in /tmp/2025acf/src/app/components/home/home.component.ts! Add HomeComponent to the NgModule to fix it.
Cannot determine the module for class HeaderComponent in /tmp/2025acf/src/app/components/header/header.component.ts! Add HeaderComponent to the NgModule to fix it.
src/app/app-routing.module.ts(3,15): error TS2304: Cannot find name 'Routes'.
```

This seems like this step is part of the setup process?

ganny26 commented:
@timothyis The steps whichever i mentioned earlier are if someone wants to create a new component they can follow but now i think it is not needed to add in *readme* so i've removed those now.
In current updated readme i've added steps to run and deploy that would be much sufficient.
if you have any thought let me know i will update accordingly 😀

msweeneydev commented:
Hey @ganny26, thank you for making this contribution. As you may have seen, we recently revamped this repository with simpler, frontend focused examples, including Angular.

As a result, I am closing this pull request. Thank you again for taking the time to contribute.
